Mathlib/GroupTheory/Perm/Sign.lean

```diff
@@ -112,14 +113,34 @@ theorem swap_induction_on [Finite α] {P : Perm α → Prop} (f : Perm α) :
         (ih _ ⟨rfl, fun v hv => hl.2 _ (List.mem_cons_of_mem _ hv)⟩ h1 hmul_swap)
 #align equiv.perm.swap_induction_on Equiv.Perm.swap_induction_on
 
-theorem closure_isSwap [Finite α] : Subgroup.closure { σ : Perm α | IsSwap σ } = ⊤ := by
+theorem mclosure_isSwap [Finite α] : Submonoid.closure { σ : Perm α | IsSwap σ } = ⊤ := by
   cases nonempty_fintype α
-  refine eq_top_iff.mpr fun x _ => ?_
+  refine top_unique fun x _ ↦ ?_
   obtain ⟨h1, h2⟩ := Subtype.mem (truncSwapFactors x).out
   rw [← h1]
-  exact Subgroup.list_prod_mem _ fun y hy => Subgroup.subset_closure (h2 y hy)
+  exact Submonoid.list_prod_mem _ fun y hy ↦ Submonoid.subset_closure (h2 y hy)
+
+theorem closure_isSwap [Finite α] : Subgroup.closure { σ : Perm α | IsSwap σ } = ⊤ :=
+  Subgroup.closure_eq_top_of_mclosure_eq_top mclosure_isSwap
 #align equiv.perm.closure_is_swap Equiv.Perm.closure_isSwap
 
+/-- Every finite symmetric group is generated by transpositions of adjacent elements. -/
+theorem mclosure_swap_castSucc_succ (n : ℕ) :
+    Submonoid.closure (Set.range fun i : Fin n ↦ swap i.castSucc i.succ) = ⊤ :=
+  top_unique <| by
```

Comment:
Unless a proof is very short, it's in general more readable to have a full tactic mode proof. So
```suggestion
    Submonoid.closure (Set.range fun i : Fin n ↦ swap i.castSucc i.succ) = ⊤ := by
  apply top_unique
```
and then unindent the rest.

Reply:
done